### PR TITLE
Add config-provided allowlist for ignore_bots instead of flat boolean on bot construction

### PR DESCRIPTION
**Pull request by gregdan3**
Merges after https://github.com/arch-community/qbot/pull/225

This one's super silly. Important changes are at bottom of modules/init.rb. Tested and works as intended.

I really like ruby's monkeypatching mechanics though! Far better than Python's, ha.

### CODE REVIEW

File: lib/init.rb [hunk 1]
```diff
@@ -78,3 +78,18 @@ def self.run!
   end
   # rubocop: enable Metrics/MethodLength, Metrics/AbcSize
 end
+
```
Comment (anna328p) [inline lib/init.rb:81]: lib/patches.rb

**Review by anna328p — CHANGES_REQUESTED** [review]
I assume you only want the one last commit in this PR, not also the three adding the voicekick command?

File: lib/init.rb (hunk 1 above)
Comment (gregdan3) [reply lib/init.rb]: Yep, my b. Fixed

**Comment by gregdan3** [comment]
> I assume you only want the one last commit in this PR, not also the three adding the voicekick command?

Corrected this.

**Review by anna328p — APPROVED** [review]

**Comment by anna328p** [comment]
oh, linter is complaining

**Comment by anna328p** [comment]
should probably suppress `Style/OptionalBooleanParameter` here because we don't want to change the API
the other lints are valid though

**Comment by gregdan3** [comment]
lints fixed!

fun bit: it also complains about having a comment on a def line, so it has to be two extra lines surrounding the def instead of one ending it.

**Comment by anna328p** [comment]
i've run into that a bunch of times too...

**Pull request merged**